core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1180,32 +1181,42 @@ public PhysicalOperation visitWindow(WindowNode node, LocalExecutionPlanContext
                         .collect(toImmutableList());
 
                 WindowFunctionSupplier windowFunctionSupplier;
-                if (function.getOrderingScheme().isPresent()) {
-                    OrderingScheme orderingScheme = function.getOrderingScheme().orElseThrow();
-                    List<Symbol> sortKeys = orderingScheme.orderBy();
-                    List<SortOrder> sortOrders = sortKeys.stream()
-                            .map(orderingScheme::ordering)
-                            .collect(toImmutableList());
-
-                    ImmutableList.Builder<Integer> sortKeysArguments = ImmutableList.builder();
-                    sortKeys.forEach(orderingArgumentSymbol -> {
-                        argumentChannels.add(source.getLayout().get(orderingArgumentSymbol));
-                        sortKeysArguments.add(argumentChannels.size() - 1); // last added argument
-                    });
+                if (resolvedFunction.functionKind() == FunctionKind.AGGREGATE) {
```

Comment:
This refactor improves readability a lot by making it clear that ordering is only supported for aggregate functions.
However, we lost one assertion: now, if a function is not aggregate, and it has the ordering scheme, we will silently ignore it. Is this case checked in the Analyzer?

Reply:
It was checked in the analyzer. But the test was missing. I added that.